deploy/kubernetes/helm/che/templates/deployment.yaml

```diff
@@ -33,8 +33,8 @@ spec:
       initContainers:
 {{- if .Values.global.multiuser }}
       - name: wait-for-postgres
-        image: alpine:3.5
-        command: ["sh", "-c", "apk --no-cache add curl jq ; adresses_length=0; until [ $adresses_length -gt 0 ]; do echo waiting for postgres to be ready...; sleep 2; endpoints=`curl -s --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt -H \"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\"     https://kubernetes.default/api/v1/namespaces/$POD_NAMESPACE/endpoints/postgres`; adresses_length=`echo $endpoints | jq -r \".subsets[]?.addresses // [] | length\"`; done;"]
+        image: eclipse/che-wait-endpoint:nightly
+        command: ["sh", "-c", "adresses_length=0; until [ $adresses_length -gt 0 ]; do echo waiting for postgres to be ready...; sleep 2; endpoints=`curl -s --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt -H \"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\"     https://kubernetes.default/api/v1/namespaces/$POD_NAMESPACE/endpoints/postgres`; adresses_length=`echo $endpoints | jq -r \".subsets[]?.addresses // [] | length\"`; done;"]
```

Comment:
could we have the command as well inside the image as entrypoint

and just provide the name of the endpoint ?
like 'postgres' for postgres endpoint and 'keycloak' for keycloak endpoint ?

it will display the name of the pod to be ready and use this name for https://kubernetes.default/api/v1/namespaces/$POD_NAMESPACE/endpoints/ as well

Reply:
@benoitf  not sure if i undersood fully, but it seems to be working this way